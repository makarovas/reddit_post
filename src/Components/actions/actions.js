import { INCREMENT, DECREMENT, ADD_FLAG } from "../consts/consts";

export function getPosts() {
  return (dispatch, getState) => {
    dispatch({ type: "GET_POSTS_BEGIN" });
    fetch("https://www.reddit.com/r/reactjs.json")
      .then(res => res.json())
      .then(json => {
        console.log(json.data.children.map(c => c.data));
        dispatch({
          type: "GET_POSTS_SUCCESS",
          posts: json.data.children.map(c => c.data)
        });
      })
      .catch(error => {
        dispatch({ type: "GET_POSTS_ERROR", error });
      });
  };
}

export function deletePost() {
  return { type: "DELETE_POST" };
}
