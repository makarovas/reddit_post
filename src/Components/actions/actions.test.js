import { deletePost } from "./actions";

test("deletePost", () => {
  //given
  const result = deletePost();
  // expect
  expect(result.type).toEqual("DELETE_POST");
  expect(result).toMatchSnapshot();
});
