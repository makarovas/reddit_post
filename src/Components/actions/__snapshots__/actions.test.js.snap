// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deletePost 1`] = `
Object {
  "type": "DELETE_POST",
}
`;
